src/chapter4/openmp.md

```diff
@@ -1 +1,99 @@
-# OpenMP
+# Parallel Computing with OpenMP
+
+## What is OpenMP
+
+OpenMP, stand for open multi-processing is an API for writing multithreaded applications
+
+It has a set of compiler directives and library routines for parallel applications, and it greatly simplifies writing multi-threaded code in Fortran, C and C++.
+
+Just few lines of additional code can make your application parallel 
+
+OpenMP uses shared memory architecture. It assumes all code runs on a single server
+
+## Threads
+
+![](src/chapter4/_attachments/Pasted%20image%2020230325111415.png)
+
+A thread of execution is the smallest instruction that can be managed independently by an operating system.
+
+In parallel region, multiple threads are spawned and utilises the cores on CPU
+
+> [!note]
+> Only one thread exists in a serial region
+
+## Compiler Directive \# pragma
+
+-   Source codes with “#” provide additional information to compiler
```

Comment:
The preprocessor will be discussed in the C section. I recommend discussion more about how it is `#pragma` are used as compiler specific preprocessor directives and then go into what it does a bit more.

Reply:
This still hasn't been resolved